airlock_processor/StatusChangedQueueTrigger/__init__.py

```diff
@@ -1,18 +1,136 @@
+from azure.mgmt.storage import StorageManagementClient
+from pydantic import BaseModel, parse_obj_as
 
 
-def main(msg: func.ServiceBusMessage,
-         outputEvent: func.Out[func.EventGridOutputEvent]):
+class RequestPropertiesData(BaseModel):
+    request_id: str
+    status: str
+    type: str
+    workspace_id: str
 
-    logging.info('Python ServiceBus queue trigger processed message: %s', msg.get_body().decode('utf-8'))
-    outputEvent.set(
-        func.EventGridOutputEvent(
-            id="step-result-id",
-            data={"tag1": "value1", "tag2": "value2"},
-            subject="test-subject",
-            event_type="test-event-1",
-            event_time=datetime.datetime.utcnow(),
-            data_version="1.0"))
+
+class RequestProperties(BaseModel):
+    data: RequestPropertiesData
+
+
+def main(msg: func.ServiceBusMessage):
+
+    body = msg.get_body().decode('utf-8')
+    logging.info('Python ServiceBus queue trigger processed message: %s', body)
+
+    try:
+        request_properties = extract_properties(body)
+
+        new_status = request_properties.data.status
+        req_id = request_properties.data.request_id
+        ws_id = request_properties.data.workspace_id
+        request_type = request_properties.data.type
+    except Exception:
+        logging.error('Failed processing request - invalid message: %s', body)
+        raise
+
+    logging.info('Processing request with id %s. new status is "%s", type is "%s"', req_id, new_status, type)
+
+    if (is_require_data_copy(new_status)):
+        logging.info('Request with id %s. requires data copy between storage accounts', req_id)
+        source_account_name, source_account_key, sa_source_connection_string, sa_dest_connection_string = get_source_dest_env_vars(new_status, request_type, ws_id)
+        blob_operations.copy_data(source_account_name, source_account_key, sa_source_connection_string, sa_dest_connection_string, req_id)
+        return
+
+    # Todo: handle other cases...
+
+
+def extract_properties(body: str) -> RequestProperties:
+    try:
+        json_body = json.loads(body)
+        result = parse_obj_as(RequestProperties, json_body)
+        if not result:
+            raise Exception("Failed parsing request properties")
+    except json.decoder.JSONDecodeError:
+        logging.error(f'Error decoding object: {body}')
+        raise
+    except Exception as e:
+        logging.error(f'Error extracting properties: {e}')
+        raise
+
+    return result
+
+
+def is_require_data_copy(new_status: str):
+    if new_status.lower() in ["submitted", "approved", "rejected", "blocked"]:
+        return True
+    return False
+
+
+def get_source_dest_env_vars(new_status: str, request_type: str, workspace_id: str):
+
+    # sanity
+    if is_require_data_copy(new_status) is False:
+        raise Exception("Given new status is not supported")
+
+    try:
+        tre_id = os.environ["TRE_ID"]
+        subscription_id = os.environ["AZURE_SUBSCRIPTION_ID"]
```

Comment:
Knowing which subscription this runs in looks odd. Why do we need it?

Reply:
This is what the ctor asks for
https://docs.microsoft.com/en-us/python/api/azure-mgmt-storage/azure.mgmt.storage.storagemanagementclient?view=azure-python#constructor